controller/konnect/index_konnectextension.go

```diff
@@ -11,6 +12,8 @@ const (
 	IndexFieldKonnectExtensionOnAPIAuthConfiguration = "konnectExtensionAPIAuthConfigurationRef"
 	// IndexFieldKonnectExtensionOnSecrets is the index field for KonnectExtension -> Secret.
 	IndexFieldKonnectExtensionOnSecrets = "konnectExtensionSecretRef"
+	// IndexFiekdKonnectExtensionOnControlPlane is the index field for KonnectExtension -> KonnectGatewayControlPlane.
+	IndexFiekdKonnectExtensionOnControlPlane = "konnectExtensionControlPlaneRef"
```

Comment:
```suggestion
	IndexFieldKonnectExtensionOnKonnectGatewayControlPlane = "konnectExtensionControlPlaneRef"
```

Reply:
How about also renaming it to `konnectExtensionKonnectGatewayControlPlaneRef`?